toxcore/bin_pack.h

```diff
@@ -34,7 +51,7 @@ typedef bool bin_pack_cb(Bin_Pack *bp, const Logger *logger, const void *obj);
  * @param arr is the object array as void pointer.
  * @param index is the index in the object array that is currently being packed.
  */
-typedef bool bin_pack_array_cb(Bin_Pack *bp, const Logger *logger, const void *arr, uint32_t index);
+typedef bool bin_pack_array_cb(const void *arr, uint32_t index, const Logger *logger, Bin_Pack *bp);
```

Comment:
Re-arrange `@param` in function doc too?

Reply:
No, I don't think so. There doesn't seem to be anything in the function doc that needs to be rearranged here.